tests/test_asyncio/test_cwe_404.py

```diff
@@ -1,147 +1,261 @@
 import asyncio
-import sys
+import contextlib
+import urllib.parse
 
 import pytest
 
 from redis.asyncio import Redis
 from redis.asyncio.cluster import RedisCluster
+from redis.asyncio.connection import async_timeout
 
 
-async def pipe(
-    reader: asyncio.StreamReader, writer: asyncio.StreamWriter, delay: float, name=""
-):
-    while True:
-        data = await reader.read(1000)
-        if not data:
-            break
-        await asyncio.sleep(delay)
-        writer.write(data)
-        await writer.drain()
+@pytest.fixture
+def redis_addr(request):
```

Comment:
@kristjanvalur WDYT about moving this fixture to conftest.py?
Now we have the exact same code 3 times (here, in [test_cluster.py](https://github.com/redis/redis-py/blob/6d32503a58d841a41556cebcf188add324c6ce86/tests/test_cluster.py#LL128C1-L129C1), and in [test_asyncio/test_cluster.py](https://github.com/redis/redis-py/blob/6d32503a58d841a41556cebcf188add324c6ce86/tests/test_asyncio/test_cluster.py#L105))

Reply:
there is already a "master_host" fixture, I'll use that instead.